solt/transforms/_transforms.py

```diff
@@ -63,24 +64,28 @@ def _apply_img(self, img: np.ndarray, settings: dict):
             return np.ascontiguousarray(img[::-1, ::-1, ...])
 
     def _apply_mask(self, mask: np.ndarray, settings: dict):
+        ensure_valid_mask(mask, num_dims=(2, ))
```

Comment:
As mentioned previous, here and elsewhere, the check must be done only in the end, to make sure that the transform returns a correct shape (3 dims for images)

Reply:
This is a completely different check and is related to compatibility of the transforms with the different input types.
I removed the one implemented previously by the decorator, and implemented it in-place for the relevant functions (see `.reshape(shape_in)`). The decorator was only fixing the implicit convertion from `(h, w, 1)` to `(h, w)` done by multiple OpenCV functions.
I think the output reshaping has to be done explicitly within every transform.

Reply:
I do not agree. From the development API point of view, it is much easier to have a function that will make sure that the shape is exactly as it should be. That function will be applied only to the output.